include/pros/optical.h

```diff
@@ -0,0 +1,265 @@
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ * \return poximity value if the operation was successful or PROS_ERR if
+ * the operation failed, setting errno.
+ */
+int32_t optical_get_proximity(uint8_t port);
+
+/**
+ * Set the pwm value of the White LED on the sensor
+ *
+ * value that ranges from 0 to 100
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ */
+int32_t optical_set_led_pwm(uint8_t port, uint8_t value);
+
+/**
+ * Get the pwm value of the White LED on the sensor
+ *
+ * value that ranges from 0 to 100
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ * \return LED pwm value if the operation was successful or PROS_ERR if
+ * the operation failed, setting errno.
+ */
+int32_t optical_get_led_pwm(uint8_t port);
+
+/**
+ * Get the processed RGBC data from the sensor
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ * \return rgb value if the operation was successful or PROS_ERR if
+ * the operation failed, setting errno.
+ */
+optical_rgb_s_t optical_get_rgb(uint8_t port);
+
+/**
+ * Get the raw un-processed RGBC data from the sensor
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ * \return raw rgb value if the operation was successful or PROS_ERR if
+ * the operation failed, setting errno.
+ */
+optical_raw_s_t optical_get_raw(uint8_t port);
+
+/**
+ * Get the most recent gesture data from the sensor
+ *
+ * Gestures will be cleared after 500mS
+ * 0 = no gesture
+ * 1 = up (towards cable)
+ * 2 = down
+ * 3 = right
+ * 4 = left
+ *
+ * This function uses the following values of errno when an error state is
+ * reached:
+ * ENXIO - The given value is not within the range of V5 ports (1-21).
+ * ENODEV - The port cannot be configured as an Optical Sensor
+ *
+ * \param port
+ *        The V5 Optical Sensor port number from 1-21
+ * \return gesture value if the operation was successful or PROS_ERR if
```

Comment:
> or PROS_ERR if

This looks like ub: https://stackoverflow.com/questions/366017/what-is-the-size-of-an-enum-in-c. Looks like you need to make PROS_ERR part of the enum.

Reply:
Practically though, if it works now it's probably fine for now, though it's a simple fix.